lib/chef/cookbook/syntax_check.rb

```diff
@@ -110,7 +110,7 @@ def remove_ignored_files(file_list)
       end
 
       def remove_uninteresting_ruby_files(file_list)
-        file_list.reject { |f| f =~ %r{#{cookbook_path}/(files|templates)/} }
+        file_list.reject { |f| f =~ %r{#{Chef::Util::PathHelper.escape_glob_dir(cookbook_path)}/(files|templates)/} }
```

Comment:
Regex escape was my first thought. It solves the `too short control escape` error, but it also breaks functionality of `remove_uninteresting_ruby_files` method.

When debugged I found out that even Windows `file_list` have a path consisting to forward slashes and, in the other hand, `cookbook_path` consists to backslashes.

So, it's necessary to convert backlashes to forward slashes. Based on code documentation, `escape_glob_dir` is a recomended way for that.
And of course, when we convert backslashed to forward slashes `too short control escape` error disappears.